docs/reference/indices/diskusage.asciidoc

```diff
@@ -0,0 +1,154 @@
+[[indices-disk-usage]]
+=== Analyze index disk usage API
+++++
+<titleabbrev>Analyze index disk usage</titleabbrev>
+++++
+
+experimental[]
+
+Analyzes the disk usage of each field of an index. This API
+might not support indices created in previous {es} versions.
+
+[source,console]
+--------------------------------------------------
+POST /my-index-000001/_index_disk_usage?run_expensive_tasks=true
+--------------------------------------------------
+// TEST[setup:my_index]
+
+
+[[analyze-index-disk-usage-api-request]]
+==== {api-request-title}
+
+`POST /<index>/_index_disk_usage`
+
+[[analyze-index-disk-usage-api-request-prereqs]]
+==== {api-prereq-title}
+
+* If the {es} {security-features} are enabled, you must have the `manage`
+<<privileges-list-indices,index privilege>> for the target index or index alias.
+
+
+[[analyze-index-disk-usage-api-path-params]]
+==== {api-path-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index]
+
+[[analyze-index-disk-usage-api-query-params]]
+==== {api-query-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
++
+Defaults to `true`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
++
+Defaults to `open`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=wait_for_active_shards]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
+
+`run_expensive_tasks`::
+(Optional, Boolean)
+Since analyzing per field disk usage of indices is expensive and resource-intensive,
+the parameter `run_expensive_tasks` must be set to `true` in order for the task to be
+performed.
+Defaults to `false`.
+
+`flush`::
+(Optional, Boolean)
+If `true`, the target index will be flushed before the analyzing process starts.
+If `false`, then uncommitted data might not be included in the result.
+Defaults to `true`.
+
+
+[[analyze-index-disk-usage-api-example]]
+==== {api-examples-title}
+
+The following request analyzes the disk usage of index `my-index-000001`.
+
+[source,console]
+--------------------------------------------------
+POST /my-index-000001/_index_disk_usage?run_expensive_tasks=true
+--------------------------------------------------
+// TEST[setup:my_index]
+
+The API returns the following response:
+
+[source,console-response]
+--------------------------------------------------
+{
+    "_shards": {
+        "total": 1,
+        "successful": 1,
+        "failed": 0
+    },
+    "my-index-000001": {
+        "_all": {
+            "total": "925.4mb",
+            "total_in_bytes": 970370607,
+            "terms": "38.6mb",
+            "terms_in_bytes": 40541738,
+            "postings": "65.7mb",
+            "postings_in_bytes": 68963453,
+            "proximity": "11.6mb",
```

Comment:
It is challenging because some bytes record information about two different things. E.g. some vints for doc IDs steal a bit to know whether the freq is greater than 1, and some vints for positions steal a bit to know whether there is a payload associated with this position.

Reply:
+1, I will add this option in a follow-up.